pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -0,0 +1,312 @@
+		}
+	}
+	// Update CSI config map
+	// If the mon endpoints change, the mon health check go routine will take care of updating the
+	// config map, so no special care is needed in this controller
+	csiClusterConfigEntry := csi.CsiClusterConfigEntry{
+		Monitors:       csi.MonEndpoints(r.clusterInfo.Monitors),
+		RadosNamespace: cephBlockPoolRadosNamespace.Name,
+	}
+	err = csi.SaveClusterConfig(r.context.Clientset, buildClusterID(cephBlockPoolRadosNamespace), r.clusterInfo, &csiClusterConfigEntry)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
+	}
+	// Success! Let's update the status
+	if cephCluster.Spec.External.Enable {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionConnected)
+	} else {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionReady)
+	}
+
+	// Return and do not requeue
+	logger.Debug("done reconciling")
+	return reconcile.Result{}, nil
+}
+
+// Create the ceph blockpool rados namespace
+func (r *ReconcileCephBlockPoolRadosNamespace) createOrUpdateRadosNamespace(cephBlockPoolRadosNamespace *cephv1.CephBlockPoolRadosNamespace) error {
+	logger.Infof("creating ceph blockpool rados namespace %s in namespace %s", cephBlockPoolRadosNamespace.Name, cephBlockPoolRadosNamespace.Namespace)
+
+	err := cephclient.CreateRadosNamespace(r.context, r.clusterInfo, cephBlockPoolRadosNamespace.Spec.BlockPoolName, cephBlockPoolRadosNamespace.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+	}
+
+	return nil
+}
+
+// Delete the ceph blockpool rados namespace
+func (r *ReconcileCephBlockPoolRadosNamespace) deleteRadosNamespace(cephBlockPoolRadosNamespace *cephv1.CephBlockPoolRadosNamespace) error {
+	logger.Infof("deleting ceph blockpool rados namespace object %q", cephBlockPoolRadosNamespace.Name)
+	if err := cephclient.DeleteRadosNamespace(r.context, r.clusterInfo, cephBlockPoolRadosNamespace.Spec.BlockPoolName, cephBlockPoolRadosNamespace.Name); err != nil {
+		return errors.Wrapf(err, "failed to delete ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+	}
+
+	logger.Infof("deleted ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+	return nil
+}
+
+// updateStatus updates an object with a given status
+func (r *ReconcileCephBlockPoolRadosNamespace) updateStatus(client client.Client, name types.NamespacedName, status cephv1.ConditionType) {
+	cephBlockPoolRadosNamespace := &cephv1.CephBlockPoolRadosNamespace{}
+	if err := client.Get(r.opManagerContext, name, cephBlockPoolRadosNamespace); err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("CephBlockPoolRadosNamespace resource not found. Ignoring since object must be deleted.")
+			return
+		}
+		logger.Warningf("failed to retrieve ceph blockpool rados namespace %q to update status to %q. %v", name, status, err)
+		return
+	}
```

Comment:
We already do this before calling `updateStatus`. Can't we just pass that same object here and issue an update?

Reply:
should be possible. @leseb any thoughts on this one?

Reply:
The get is there to make sure we fetch the latest version of the object before updating it. I have a PR I'll rebase on top of this, to use a better pattern: https://github.com/rook/rook/pull/9686

Reply:
Isn't it problematic to update status on latest version even before operator reconciles it? `retryOnConflict` makes it easier to try updating the same object version that the operator reconciled.